### PR TITLE
Featured Images on Posts! (with Thumbnails)

### PR DESCRIPTION
**Pull request by coopermaruyama**
_1st commit with featured image so probably has bugs, please let me know about any._

 - Adds an input to edit post page to upload a featured image for a post
 - Uses S3 upload if enabled (needs refactoring)
 - Reads dimensions of image and stores in db
 - Option to use featured image as background for title
 - Subtle styling changes inspired by medium.com
 - Utility method added which sets a variable and changes the class of
   the featured image's container if the window width is bigger than
   the image width.
 - Blog index shows featured image as thumbnail if exists.

### CODE REVIEW

**Comment by coopermaruyama** [comment]
@hellogerard any feedback on this PR?

File: client/views/blog/blog.html [hunk 1]
```diff
@@ -14,16 +14,27 @@
...
+          </h2>
+          <div class="post-meta">
+            By {{authorName}} on {{blogFormatDate publishedAt}}
+            {{> blogTags }}
+          </div>
+          <div class="the-content">
+            <small>{{{excerpt}}}</small>
+          </div>
+        </div>
+        {{#if featuredImage}}
```
Comment (rgoomar) [inline client/views/blog/blog.html]: Does this mean the featuredImage is below the title and the excerpt in mobile? Wouldn't you want it above?

**Comment by coopermaruyama** [comment]
@rgoomar you're right, I flipped it around so it's above the title now.

File: client/views/admin/edit.coffee [hunk 2]
```diff
@@ -170,6 +191,51 @@ Template.blogAdminEdit.events
...
+        #   xhr = new XMLHttpRequest()
+        #   xhr.upload.onprogress = that.updateProgressBar
+        #   xhr
+        cache: false
+        contentType: false
+        complete: (jqxhr) ->
+          if post.id? then post.update { featuredImage: "/fs/#{id}" }
+          return
+        processData: false
+        data: that.options.formatData(file)
```
Comment (hellogerard) [inline client/views/admin/edit.coffee]: @coopermaruyama Getting a `Uncaught ReferenceError: that is not defined` on this line.

**Comment by coopermaruyama** [comment]
Ah, I totally overlooked the local file store and forgot to test it with the featured image uploader. This new commit should take care of that. Let me know if you still have issues.

**Pull request merged**